src/StripeTests/BaseStripeTest.cs

```diff
@@ -0,0 +1,158 @@
+namespace StripeTests
+{
+    using System;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+
+    using Stripe;
+
+    public class BaseStripeTest : IDisposable
+    {
+        /// <value>Minimum required version of stripe-mock</value>
+        /// <remarks>
+        /// If you bump this, don't forget to bump `STRIPE_MOCK_VERSION` in appveyor.yml as well.
+        /// </remarks>
+        private const string MockMinimumVersion = "0.19.0";
+
+        private static string port;
+
+        private string originalApiBase;
```

Comment:
since we are moving tests to a new package, they will all be based on stripe-mock (or in the future stubbed). Can we get rid of all this in that case? Or are the test suites ran in parallel and so they conflict?

Reply:
I think there's _some_ value in cleaning up (though tbh we could simply restore the default values instead of saving the original values).

The tests are run in parallel on AppVeyor (not sure why they aren't on our laptops -- xUnit uses the "number of virtual CPUs in the PC" by default, I guess that xUnit only sees 1 for some reason). It should still work because all the tests in this project will use the same global state values, though ideally we shouldn't rely on this. A better solution would be to be able to provide all options directly in each tests, via `RequestOptions` or an hypothetical `StripeClient`, but that's not possible right now.

Reply:
> xUnit uses the "number of virtual CPUs in the PC" by default
Interesting. Never knew that.

You could create a `TestRequestOptions` of sorts which has properties for api base, upload base, and api key (private strings on BaseStripeTest now). Have the constructor of BaseStripeTest take a parameter of type `TestRequestOptions`. If the test doesn't provide this object, have it use the default set.

Reply:
That sounds like a good idea if a specific test needs to use different values, but from past experience from the other language libraries, I don't anticipate that being the case. In other libraries, tests fall into one of two categories:
- most use stripe-mock
- some use stubbed requests (and thus don't care about values for `ApiBase` etc. since they don't make network requests)